_posts/2016-03-25-hello-squabble.md

```diff
@@ -0,0 +1,49 @@
+---
+[Avid readers of our blog](/2015/04/01/building-our-site.html) know that it's built on the amazing and lovely open-source static site generator known as [Jekyll](http://jekyllrb.com/).
+
+Jekyll is a great solution for a site like ours because we get an always-on[^1], super fast, database-free site that's also capable of many of the things you'd typically associate with a database-driven site: tagging, content organization, advanced templating, etc. And best of all it's one less database-driven application for us to maintain, keep up to date with the latest security patches, and so on.
+
+However, as much as we like static HTML, we also care about you, dear reader, and we wanted to provide an easy way for you all to comment and contribute to our site's content.
+
+## How can I add comments to a static website?
+
+Most people solve this problem in Jekyll by outsourcing the comment hosting to [Disqus](https://disqus.com/). The way this works is: you embed some JavaScript in our template, and that JavaScript loads an interface to the Disqus commenting platform. Users leave a comment via the Disqus, and on subsequent page loads, Disqus finds any comments that are added for a particular URL and displays them to the end user.
+
+Disqus works very well, but there are significant privacy concerns associated with the service as they track your activity while you browse sites across the web. (This is somewhat mitigated by this [semi-hidden setting](https://help.disqus.com/customer/portal/articles/1657951) that lets you opt out of some of their tracking, but I'm willing to venture that the vast majority of Disqus users don't know about this.) It's also not especially attractive, and the pre-built styling associated with Disqus doesn't mesh with the rest of our site.
+
+## Creating an alternative
+
+So, we decided to write our alternative[^2]. Our requirements are simple: we're not going to track you across the web and try to sell you things you don't need. We're not going to offload your data to shady third parties. We just you to be able to leave a comment on a blog post!
+
+The basic plan for implementing our Disqus alternative is similar to what Disqus does: we need to write a PHP application with an API, and that API should allow us to create, delete, and access comments — either all comments or just the comments for a particular blog post.
+
+### Built on Lumen
+
+Looking around the open source landscape for the right tool to build this app, I came across [Lumen](https://lumen.laravel.com/), a slimmed down version of the well-known [Laravel](https://www.laravel.com) project. Lumen was very simple to get up and running with thanks to the [clear documentation examples](https://lumen.laravel.com/docs/5.2).
+
+The entry point of our application are some [routing rules](https://github.com/savaslabs/squabble/blob/master/app/Http/routes.php) that determine what data to give back to the user when they visit a particular API endpoint.
```

Comment:
More of a personal preference thing, but I like to link to commit hash instead of a branch, since the code could change on the branch. Same with the link below. Take it or :leaves: it.